src/bitcoin-cli.cpp

```diff
@@ -267,21 +263,17 @@ static int8_t NetworkStringToId(const std::string& str)
     return UNKNOWN_NETWORK;
 }
 
-/** Class that handles the conversion from a command-line to a JSON-RPC request,
+/** Handle the conversion from a command-line to a JSON-RPC request,
  * as well as converting back to a JSON object that can be shown as result.
  */
-class BaseRequestHandler
-{
-public:
+struct BaseRequestHandler {
     virtual ~BaseRequestHandler() = default;
     virtual UniValue PrepareRequest(const std::string& method, const std::vector<std::string>& args) = 0;
     virtual UniValue ProcessReply(const UniValue &batch_in) = 0;
 };
 
 /** Process addrinfo requests */
-class AddrinfoRequestHandler : public BaseRequestHandler
-{
-public:
+struct AddrinfoRequestHandler : BaseRequestHandler {
```

Comment:
simple structs are fine, but I'm not a fan of struct inheritance, I like the previous version more  in the non-trivial cases

Reply:
If I'm understanding the comment ("simple structs are fine"), it comes from the idea of structs being for POD, say, in C. In C++, classes and structs are identical except for their default inheritance and member access levels, and for the use cases here, a struct appears to be the more appropriate choice that allows simpler, cleaner code (see also [stackoverflow.com/a/46339933](https://stackoverflow.com/a/46339933). Struct inheritance is also fine (see discussions like [stackoverflow.com/questions/3574040/c-can-a-struct-inherit-from-a-class](https://stackoverflow.com/questions/3574040/c-can-a-struct-inherit-from-a-class). If there is a technical issue beyond style nits, happy to re-evaluate.